types/metro-resolver/errors.d.ts

```diff
@@ -0,0 +1,37 @@
+import { FileAndDirCandidates, FileCandidates } from './types';
+
+export class FailedToResolvePathError extends Error {
+    candidates: FileAndDirCandidates;
+}
+
+export class FailedToResolveNameError extends Error {
+    dirPaths: ReadonlyArray<string>;
+    extraPaths: ReadonlyArray<string>;
+}
```

Comment:
We should use `declare class` here, and also include the contructor:
```suggestion
export declare class FailedToResolveNameError extends Error {
    dirPaths: ReadonlyArray<string>;
    extraPaths: ReadonlyArray<string>;
    constructor(dirPaths: ReadonlyArray<string>, extraPaths: ReadonlyArray<string>);
}
```